src/com/hms/flexyosisoftconnector/TagPayload.java

```diff
@@ -0,0 +1,37 @@
+package com.hms.flexyosisoftconnector;
+
+import com.hms_networks.americas.sc.datapoint.DataPoint;
+
+/**
+ * This class holds the incomplete payload as well as one of the datapoint objects.
+ *
+ * @author HMS Networks Inc. Solution Center
+ */
+public class TagPayload {
```

Comment:
string builder is used to construct the string, and this is only set once when the payload string is complete. The size limits are set in the OSIsoftJsonPayload class to be 200 data points worth of payload. I will swap this to string builder lite since it may be huge

Reply:
Then why is there an add method and not just a method like

  public void setPayload(String newPayload) {
    payload = newPayload;
  }

Reply:
Is this class just for one datapoint? if so I think the class description should be revised.

Reply:
oh my mistake, yeah this did have a few chunks added in. forgot that i switched it.  osisoft json payload builds a new part of the payload and gives it to this class. it was just to keep the tag information coupled with the payload, so when the tag is first encountered it sets the data point once. Some information stored in the data point object is needed later on. It is for a single tag and a single payload, many data points. But yes, it should be string builder instead of string

Reply:
I will make that change after everything else is done